qiskit/transpiler/passes/routing/sabre_swap.py

```diff
@@ -332,9 +386,21 @@ def _obtain_swaps(self, front_layer, current_layout):
                     virtual_neighbor = current_layout[neighbor]
                     swap = sorted([virtual, virtual_neighbor], key=lambda q: self._bit_indices[q])
                     candidate_swaps.add(tuple(swap))
-
         return candidate_swaps
 
+    def _add_greedy_swaps(self, front_layer, dag, layout, qubits):
+        """Mutate ``dag`` and ``layout`` by applying greedy swaps to ensure that at least one gate
+        can be routed.  Returns the gate (as a :class:`.DAGOpNode`) that can now be routed."""
+        layout_map = layout._v2p
+        target_node = min(
+            front_layer,
+            key=lambda node: self.dist_matrix[layout_map[node.qargs[0]], layout_map[node.qargs[1]]],
+        )
+        for pair in _shortest_swap_path(tuple(target_node.qargs), self.coupling_map, layout):
+            self._apply_gate(dag, DAGOpNode(op=SwapGate(), qargs=pair), layout, qubits)
+            layout.swap(*pair)
```

Comment:
so this routes one qubit to another by applying gates on the path one-by-one? so something like this?
```
----x-----------
    |
----x----x----
         |
---------x---x--
             |
-------------x--
```

typically it is better to do a meet-in-the-middle approach to save depth, like this:
```
----x-----------
    |
----x----x----
         |
----x----x------
    |
----x-------------
```

Reply:
Thanks - I got so wrapped up in having solved the problem I forgot to even think about the depth!  Fixed in 5b238ee07.